File: frontend/src/features/party/models/partyListItem.ts
export interface PartyListItem {
  partyId: number;
  name: string;
  ceo: string;
  manager: string;
  contact: string;
  email: string;
}

export interface PartySearchParams {
  page: number;
  size: number;
  name?: string;
  sort?: string;
}

export function mapPartyListItem(input: any): PartyListItem {
  return {
    partyId: Number(input?.partyId ?? 0),
    name: String(input?.name ?? ''),
    ceo: String(input?.ceo ?? ''),
    manager: String(input?.manager ?? ''),
    contact: String(input?.contact ?? ''),
    email: String(input?.email ?? ''),
  };
}

File: abms-application/src/main/java/kr/co/abacus/abms/application/party/dto/PartyListItem.java
package kr.co.abacus.abms.application.party.dto;

import org.jspecify.annotations.Nullable;

public record PartyListItem(
        Long partyId,
        String name,
        @Nullable String ceo,
        @Nullable String manager,
        @Nullable String contact,
        @Nullable String email) {

}
